File: 0739-daily-temperatures/0739-daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        var stk = new Stack<int>(); // storing days
        var res = new int[temperatures.Length];
        for(int i = 0; i < temperatures.Length; i++)
        {
            if(stk.Count == 0 || temperatures[i] <= temperatures[stk.Peek()])
            {
                stk.Push(i);
            }
            else
            {
                while(stk.Count > 0 && temperatures[i] > temperatures[stk.Peek()])
                {
                    res[stk.Peek()] = i - stk.Peek();
                    stk.Pop();
                }
                stk.Push(i);
            }
        }
        return res;
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.swift
class Solution {
    func dailyTemperatures(_ temperatures: [Int]) -> [Int] {
        var res: [Int] = Array(repeating: 0, count: temperatures.count)
        var stack = [Int]()

        stack.reserveCapacity(res.count)

        for i in 0..<temperatures.count {

            while !stack.isEmpty && temperatures[stack.last!] < temperatures[i] {
                let last = stack.removeLast()
                res[last] = i - last
            }

            stack.append(i)
        }

        return res
    }
}